src/server/pkg/hashtree/cache.go

```diff
@@ -115,6 +125,7 @@ func (c *Cache) GetOrAdd(key interface{}, generator func() (HashTree, error)) (H
 
 		return nil, err
 	}
+	newValue.GetRef()
```

Comment:
I think since the tree comes with a default reference, this isn't needed?

Reply:
The default reference covers the fact that the tree is in the cache, this one covers the fact that it's returned from `GetOrAdd`. Otherwise on the initial cache miss the caller would decrement the refs to 0 and destroy the newly cached tree.